### PR TITLE
Add apple metadata cache

### PR DESCRIPTION
**Pull request by amCap1712**
See #2546 and #2566.

### CODE REVIEW

**Comment by amCap1712** [comment]
1. For insert logic to work correctly, need to unify the names of the id columns (spotify_id and apple_id).
2. Need to add a new container to run this cache.

File: listenbrainz/metadata_cache/apple/handler.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        album_ids = set()
+        for storefront in storefronts:
+            response = client.get(f"https://api.music.apple.com/v1/catalog/{storefront}/charts", {
+                "types": "albums",
+                "limit": 200
+            })
+            album_ids |= {album["id"] for album in response["results"]["albums"][0]["data"]}
+        return list(album_ids)
+
+    def get_items_from_listen(self, listen) -> list[JobItem]:
```
Comment (mayhem) [inline listenbrainz/metadata_cache/apple/handler.py:47]: docstrings

**Review by mayhem — APPROVED** [review]
Some functions are missing docstrings and then the invocation copypasta... otherwise great!

File: listenbrainz/metadata_cache/apple/handler.py (hunk 1 above)
Comment (amCap1712) [reply listenbrainz/metadata_cache/apple/handler.py:47]: fixed.

**Pull request merged**